core/store.go

```diff
@@ -315,17 +341,25 @@ func delOperation(k string) bool {
 	}
 
 	if obj, ok := store[ptr]; ok {
-		delete(store, ptr)
-		delete(expires, obj)
-		delete(keypool, k)
-		KeyspaceStat[0]["keys"]--
-
-		WatchChannel <- WatchEvent{k, "DEL", obj}
-		return true
+		if success := executeDelete(ptr, obj, k); success {
+			WatchChannel <- WatchEvent{k, "DEL", obj}
+			return true
+		}
 	}
 	return false
 }
 
+func executeDelete(ptr unsafe.Pointer, obj *Obj, key string) bool {
+	if obj == nil {
+		return false
+	}
+	delete(store, ptr)
+	delete(expires, obj)
+	delete(keypool, key)
+	KeyspaceStat[0]["keys"]--
```

Comment:
@JyotinderSingh Do we need a lock for `KeyspaceStat`? It's being used without a lock in other places as well.

Reply:
Yes, must have overlooked this. We should put it under a lock. It can be dealt with as a separate issue.

Reply:
Also, keyspace stat is only updated in write scenarios, and presently no writes happen concurrently (we are only concurrent in reads due to QWATCH).
However this may become a concern as we move towards a multithreaded model.

Reply:
Sounds good.